static/src/javascripts/projects/common/modules/identity/wizard.js

```diff
@@ -31,25 +37,85 @@ const updateCounter = (wizardEl: HTMLElement): Promise<void> =>
+        .read(() => [
+            wizardEl.getBoundingClientRect().top - 20,
+            parseInt(
+                wizardEl.dataset.position ? wizardEl.dataset.position : -1,
+                10
+            ),
+            [...wizardEl.getElementsByClassName(stepClassname)],
+        ])
+        .then(
+            (
+                [
+                    offsetTop: number,
+                    currentPosition: number,
+                    steps: Array<HTMLElement>,
+                ]
+            ) =>
+                fastdom.write(() => {
+                    if (newPosition < 0 || !steps[newPosition]) {
+                        throw new Error('Invalid position');
+                    }
+                    if (currentPosition > -1 && window.scrollY > offsetTop) {
+                        scrollTo(offsetTop, 250, 'linear');
+                    }
+                    const transitionClassnames = [
+                        stepInClassname,
+                        stepInReverseClassname,
+                        stepOutClassname,
+                        stepOutReverseClassname,
+                    ];
+                    steps.forEach((stepEl: HTMLElement, i: number) => {
+                        switch (i) {
+                            case newPosition:
+                                stepEl.classList.remove(
+                                    stepHiddenClassname,
+                                    ...transitionClassnames
+                                );
+                                wizardEl.style.minHeight = `${stepEl.getBoundingClientRect()
+                                    .height}px`;
+                                if (currentPosition > -1) {
+                                    stepEl.classList.add(
+                                        currentPosition < newPosition
+                                            ? stepInClassname
+                                            : stepInReverseClassname
+                                    );
+                                }
+                                break;
+                            case currentPosition:
+                                stepEl.classList.remove(
+                                    ...transitionClassnames
+                                );
+                                stepEl.classList.add(
+                                    ...[
+                                        stepHiddenClassname,
+                                        currentPosition < newPosition
+                                            ? stepOutClassname
+                                            : stepOutReverseClassname,
+                                    ]
+                                );
+                                setTimeout(() => {
+                                    stepEl.classList.remove(
+                                        ...transitionClassnames
+                                    );
+                                }, 200);
```

Comment:
this is a race condition right there but the way the css and the transition composition is set up it doesn't really lead to any inconsistent/empty states